include/envoy/upstream/health_checker.h

```diff
@@ -99,6 +99,21 @@ class HealthCheckEventLogger {
    */
   virtual void logAddHealthy(envoy::data::core::v2alpha::HealthCheckerType health_checker_type,
                              const HostDescriptionConstSharedPtr& host, bool first_check) PURE;
+
+  /**
+   * Log a degraded healthy host event.
+   * @param health_checker_type supplies the type of health checker that generated the event.
+   * @param host supplies the host that generated the event.
+   */
+  virtual void logDegraded(envoy::data::core::v2alpha::HealthCheckerType health_checker_type,
+                           const HostDescriptionConstSharedPtr& host) PURE;
+  /**
+   * Log a no degraded healthy host event.
+   * @param health_checker_type supplies the type of health checker that generated the event.
+   * @param host supplies the host that generated the event.
+   */
+  virtual void logNoMoreDegraded(envoy::data::core::v2alpha::HealthCheckerType health_checker_type,
```

Comment:
```suggestion
  virtual void logNoLongerDegraded(envoy::data::core::v2alpha::HealthCheckerType health_checker_type,
```

Seems a bit less clunky?